airflow/task/context/execution_context.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from types import ModuleType
+from typing import Any, NamedTuple
+
+from pendulum import Pendulum
+
+from airflow.exceptions import AirflowException
+
+
+class ExecutionContext(NamedTuple):
+    """
+    Defines all stateful properties during execution.
+    This context can be passed to any task either via task.context.current.get_current_context, or via
+    requesting specific properties through the task's ctor, as in:
+
+        def f(execution_date):
+            print(execution_date) # This property is filled automatically by airflow
+    """
+    conf: Any
```

Comment:
Main problem: Using `Any` instead of the actual type. Circular dependencies forbid us from using the correct typing here, would appreciate ideas on improving this

Reply:
We can try this approach:
https://github.com/databand-ai/airflow/pull/3#discussion_r414481338